wpi_interface/src/main/java/coppercore/wpi_interface/MonitoredSubsystem.java

```diff
@@ -0,0 +1,47 @@
+package coppercore.wpi_interface;
+
+import coppercore.monitors.Monitor;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import java.util.ArrayList;
+import java.util.List;
+import org.littletonrobotics.junction.Logger;
+
+public class MonitoredSubsystem extends SubsystemBase {
+    private List<Monitor> registeredMonitors = new ArrayList<Monitor>();
+
+    public void addMonitor(Monitor monitor) {
+        registeredMonitors.add(monitor);
+    }
+
+    @Override
+    public void periodic() {
+        monitoredPeriodic();
+        runMonitors();
+    }
+
+    /**
+     * OVERRIDE ME! This function is called every time the subsystem's periodic function is called.
+     * However, MonitoredSubsytem automatically checks monitors during every periodic run.
+     * Therefore, this method should be overridden as a replacement for the normal periodic function
+     * in the implementation of the subsystem.
```

Comment:
Could we make MonitoredSubsystem an abstract class, and then force all users of MonitoredSubsystem to provide this implementation?